lib/text/ui_text_displayer.js

```diff
@@ -504,6 +508,19 @@ shaka.text.UITextDisplayer = class {
       regionElement.style.left = region.viewportAnchorX -
         region.regionAnchorX * region.width / 100 + viewportAnchorUnit;
     }
+    if (region.heightUnits !== pixelUnit &&
+        region.widthUnits !== pixelUnit &&
+        region.viewportAnchorUnits !== pixelUnit) {
+      // Clip region
+      const top = parseInt(regionElement.style.top.slice(0, -1), 10) || 0;
+      const left = parseInt(regionElement.style.left.slice(0, -1), 10) || 0;
+      const height = parseInt(regionElement.style.height.slice(0, -1), 10) || 0;
+      const width = parseInt(regionElement.style.width.slice(0, -1), 10) || 0;
+      const realTop = Math.min(100 - height, top);
```

Comment:
These should probably also do `Math.max(0, `, right?
In case a region is outside the window, in the other direction.

Reply:
Done!